### ISSUE TITLE
Version 2.0.2 requires deleted_at in unspecified models

### ISSUE THREAD

**Issue by sporto** [rubysherpas/paranoia#102]
Updated from 2.0.1 to 2.0.2, suddenly we are getting errors asking for deleted_at in models that do not include paranoia e.g.

```
PG::UndefinedColumn: ERROR: column core_permissions.deleted_at does not exist
```

Is the gem including itself automatically? Or maybe requiring that associations also has delete_at? I haven't look too much into why this is happening, but for now we are just reverting back to 2.0.1

I will try to investigate this if it isn't anything obvious to you.

**Comment by radar** [comment]
The gem should not be automatically hooking itself in anywhere. The only place I can think of this happening without an explicit call in the model is if your model inherits from a paranoid model.

If you're able to give me a simple test app which reproduces this, I would be very thankful and should know what the cause is right away.

**Comment by lappi-lynx** [comment]
Just got the same issue. Reverted back to 2.0.1 version helps.

```
Mysql2::Error: Unknown column 'people.deleted_at' in 'where clause': SELECT COUNT(*) FROM `people` INNER JOIN `student_course_histories` ON `people`.`id` = `student_course_histories`.`student_id` WHERE `people`.`deleted_at` IS NULL AND `student_course_histories`.`course_id` = 9999

# It happens when calling @course.students in controller
# Below code from models:
# models/course.rb
has_many :student_course_histories
has_many :students, through: :student_course_histories

# models/student_course_history.rb
acts_as_paranoid
```

Hope it will help to find a bug.

**Comment by radar** [comment]
@Extazystas Thanks. Without an application that can reproduce this problem I am going to be unable to fix it. Can someone please provide this?

**Comment by anilmaurya** [comment]
I faced this issue in 2.0.2 version and its specific to has_many through relationship.

**Comment by radar** [comment]
@anilmaurya As per my request above.

**Comment by tolhzar** [comment]
I'm experiencing the same issue with has_many :through association. It looks like that the issue might be connected with rails default scope implementation. I'm not sure about this, but I am 100% sure why it works in 2.0.1 and doesn't work in 2.0.2. Please see below:
2.0.2
self.paranoia_column = options[:column] || :deleted_at
default_scope { where(paranoia_column => nil) }
2.0.1
default_scope { where(self.quoted_table_name + '.deleted_at IS NULL') }

For some reason default_scope is being applied to the wrong model when the table name is not specified. Unfortunately, I don't have enough time to dig deeper now, but I hope I have given a clue.

**Comment by radar** [comment]
@rocketblr I think you've hit the nail on the head there. We should change `paranoia_column` to be defined like this:

```
self.paranoia_column = options[:column] || self.quoted_table_name + '.deleted_at'
```

Could you please submit a patch for this?

**Comment by radar** [comment]
Fixed by #164.
